--- a/src/igb_main.c
+++ b/src/igb_main.c
@@ -7366,7 +7366,12 @@ static inline void igb_rx_hash(struct igb_ring *ring,
 			       struct sk_buff *skb)
 {
 	if (netdev_ring(ring)->features & NETIF_F_RXHASH)
+#if ( LINUX_VERSION_CODE < KERNEL_VERSION(3,14,0) )
 		skb->rxhash = le32_to_cpu(rx_desc->wb.lower.hi_dword.rss);
+#else
+		skb_set_hash(skb, le32_to_cpu(rx_desc->wb.lower.hi_dword.rss),
+			     PKT_HASH_TYPE_L3);
+#endif
 }
 
 #endif
