--- igb-5.3.3.2/src/kcompat.h~	2015-09-11 23:47:20.000000000 +0200
+++ igb-5.3.3.2/src/kcompat.h	2015-10-01 08:16:35.598629802 +0200
@@ -4576,7 +4576,7 @@
 #define HAVE_NDO_BRIDGE_GETLINK_NLFLAGS
 #endif /* 4,1,0 */
 
-#if ( LINUX_VERSION_CODE < KERNEL_VERSION(4,2,0) )
+#if ( (LINUX_VERSION_CODE < KERNEL_VERSION(4,2,0)) && !(LINUX_VERSION_CODE >= KERNEL_VERSION(4,1,9)) )
 #if (!(SLE_VERSION_CODE && SLE_VERSION_CODE >= SLE_VERSION(12,1,0)))
 static inline bool page_is_pfmemalloc(struct page *page)
 {
@@ -4589,8 +4589,10 @@
 #endif /* !SLES12sp1 */
 
 #else
-#define HAVE_NDO_DFLT_BRIDGE_GETLINK_VLAN_SUPPORT
 #undef HAVE_STRUCT_PAGE_PFMEMALLOC
 #endif /* 4.2.0 */
+#if ( LINUX_VERSION_CODE >= KERNEL_VERSION(4,2,0) )
+#define HAVE_NDO_DFLT_BRIDGE_GETLINK_VLAN_SUPPORT
+#endif /* 4.2.0 */
 
 #endif /* _KCOMPAT_H_ */
